phys2bids/physio_obj.py

```diff
@@ -594,10 +592,25 @@ def auto_trigger_selection(self):
                                 'Please run phys2bids specifying the -chtrig argument.')
             else:
                 self.trigger_idx = indexes[0]
-                LGR.info(f'{self.ch_name[self.trigger_idx]} selected as trigger channel')
         else:
-            raise Exception('No trigger channel automatically found. Please run phys2bids '
-                            'specifying the -chtrig argument.')
+            # Time-domain automatic trigger detection
+
+            # Create numpy array with all channels (excluding time)
+            channel_ts = np.array(self.timeseries[1:])
+
+            # Normalize each signal to [0,1]
+            min_ts = np.min(channel_ts, axis=1)[:, None]
+            max_ts = np.max(channel_ts, axis=1)[:, None]
+            channel_ts = (channel_ts - min_ts) / (max_ts - min_ts)
+
+            # Compute distance to the closest signal limit (0 or 1)
+            distance = np.minimum(abs(channel_ts - 0), abs(channel_ts - 1))
+            distance_mean = np.mean(distance, axis=1)
+
+            # Set the trigger as the channel with the smallest distance
+            self.trigger_idx = np.argmin(distance_mean) + 1
```

Comment:
I would suggest using np.nanargmin here just to guard against the edge case where an input only has a single value (e.g. if the equipment fails).